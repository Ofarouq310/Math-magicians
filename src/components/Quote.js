import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner } from '@fortawesome/free-solid-svg-icons';

export default function Quote() {
  const [quotes, setQuotes] = React.useState([]);
  const [isLoading, setIsLoading] = React.useState(false);
  const [hasError, setHasError] = React.useState(false);

  const apiKey = 'QJU4yxMgy1FD96wF65IqoQ==Z4VsyCU2v0Sg2qxc';

  React.useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const res = await fetch('https://api.api-ninjas.com/v1/quotes?category=happiness', {
          mode: 'cors',
          method: 'GET',
          headers: {
            'X-Api-Key': apiKey,
            'Content-Type': 'application/json',
          },
        });
        const json = await res.json();
        setQuotes(json);
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, [setQuotes, setIsLoading, setHasError]);

  const displayQuote = quotes.map((quote) => (
    <>
      <p key={1}>
        {quote.quote}
      </p>
      <p key={2}>
        -
        {' '}
        {quote.author}
      </p>
    </>
  ));

  if (hasError) {
    return (
      <div className="wrap">
        <p className="error">Something went wrong!</p>
      </div>
    );
  }
  if (isLoading) {
    return (
      <div className="loading-icon--wrap">
        <FontAwesomeIcon icon={faSpinner} spinPulse size="3x" />
      </div>
    );
  }

  return (
    <div className="quotes--page">
      <div className="quote">
        {displayQuote}
      </div>
    </div>
  );
}
