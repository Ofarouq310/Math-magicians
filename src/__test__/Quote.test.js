import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Quote from '../components/Quote';

describe('check Quote', () => {
  const tree = renderer.create(<Router><Quote /></Router>).toJSON();
  test('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});