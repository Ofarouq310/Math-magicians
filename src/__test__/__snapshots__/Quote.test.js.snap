// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check Quote renders correctly 1`] = `
<section
  className="quote"
>
  <p>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.
  </p>
  <p>
    - William Paul Thurston
  </p>
</section>
`;
